app/models/KafkaMessageReader.scala

```diff
@@ -0,0 +1,109 @@
+package models
+
+import java.io.{ByteArrayInputStream, InputStreamReader}
+import java.util.zip.{GZIPInputStream, ZipException}
+
+import com.google.common.io.CharStreams
+import kafka.api.{FetchRequestBuilder, PartitionMetadata, TopicMetadataRequest}
+import kafka.consumer.SimpleConsumer
+import play.api.Play.current
+import play.api.{Logger, Play}
+
+/**
+  * Contains code to read messages from kafka.
+  */
+object KafkaMessageReader {
+
+  private val Kafkas = Play.configuration.getString("capillary.kafkas").getOrElse("127.0.0.1:9092")
+  private val KafkaHosts = Kafkas.split(",")
+
+  private val Timeout = 100000
+  private val BufferSize = 10000000
+  private val ClientId = "Capillary"
+
+  def getMessageContent(topic: String, partition: Int, offset: Long): String = {
+    findLeader(KafkaHosts, topic, partition) match {
+      case Some(pm) => readMessage(pm, topic, offset)
+      case None => s"Could not find any leader for partition: $partition"
+    }
+  }
+
+  private def findLeader(kafkaHosts: Array[String], topic: String, partition: Int): Option[PartitionMetadata] = {
+    var simpleConsumer: Option[SimpleConsumer] = None
+    try {
+      for (kafkaHost <- kafkaHosts) {
+        val hostAndPort = kafkaHost.split(":")
+        val host = hostAndPort(0)
+        val port = hostAndPort(1).toInt
+        simpleConsumer = Some(new SimpleConsumer(host, port, Timeout, BufferSize, ClientId))
+        val topicMetadataRequest = new TopicMetadataRequest(Seq(topic), 0)
+        val topicMetadataResponse = simpleConsumer.get.send(topicMetadataRequest)
+        for (topicMetadata <- topicMetadataResponse.topicsMetadata) {
+          for (partitionMetadata <- topicMetadata.partitionsMetadata) {
+            if (partitionMetadata.partitionId == partition) {
+              Logger.debug(s"FoundPartitionMetadata, p=$partitionMetadata")
+              // scalastyle:off return
+              return Some(partitionMetadata)
+              // scalastyle:on return
+            }
+          }
+        }
+      }
+    } finally {
+      if (simpleConsumer.isDefined) {
+        simpleConsumer.get.close()
+      }
+    }
+
+    Logger.error("No partition found")
+    None
+  }
+
+  private def readMessage(partitionMetadata: PartitionMetadata, topic: String, offset: Long): String = {
+    val leadBroker = partitionMetadata.leader.get
```

Comment:
Would be nice to handle this safely.  I believe we've hit some time-consuming bugs due to directly dereferencing an optional value in the queue stuff, so I think this is an area where we will gain time by being a bit more disciplined.

I haven't looked at the code/docs for `PartitionMetadata` lately, though, nor traced the logic completely here so if this call happens to be safe due to some precondition please ignore (though in that case I think an explanatory comment would be in order). :)